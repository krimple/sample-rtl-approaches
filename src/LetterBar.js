import PropTypes from 'prop-types';
import Letter from './Letter';

export default function LetterBar ({gameLetters, usedLetterIndexes, addLetter}) {
  return (
    <div className="row justify-content-center">
      {
        gameLetters.map((character, index) => (
              <Letter
                key={index}
                index={index}
                disabled={usedLetterIndexes.indexOf(index) > -1}
                character={character}
                addLetter={addLetter}/>
        ))
      }
    </div>
  );
}

LetterBar.propTypes = {
  gameLetters: PropTypes.arrayOf(PropTypes.string),
  usedLetterIndexes: PropTypes.arrayOf(PropTypes.number),
  addLetter: PropTypes.func
};
