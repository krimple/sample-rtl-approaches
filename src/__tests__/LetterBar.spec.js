import { render, screen } from '@testing-library/react';
import LetterBar from '../LetterBar';

describe('<LetterBar />', () => {
  const props = {
    mode: 'game',
    gameLetters: ['A', 'B', 'C'],
    addLetter: jest.fn()
  };

  it('matches snapshot with one 0 used letters', () => {
    render(<LetterBar {...props} usedLetterIndexes={[]}/>);
    const buttonList = screen.queryAllByRole('button');
    expect(buttonList).toMatchSnapshot();
    expect(buttonList[0].disabled).toBe(false);
    expect(buttonList[1].disabled).toBe(false);
    expect(buttonList[2].disabled).toBe(false);
  });

  it('matches snapshot with A and C used letters', () => {
    render(<LetterBar {...props} usedLetterIndexes={[0, 2]}/>);
    const buttonList = screen.queryAllByRole('button');
    expect(buttonList).toMatchSnapshot();
    expect(buttonList[0].disabled).toBe(true);
    expect(buttonList[1].disabled).toBe(false);
    expect(buttonList[2].disabled).toBe(true);
  });
});

