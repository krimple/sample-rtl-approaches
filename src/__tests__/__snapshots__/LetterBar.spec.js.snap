// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LetterBar /> matches snapshot with A and C used letters 1`] = `
Array [
  <button
    class="btn letter"
    data-testid="letter-button"
    disabled=""
  >
    A
  </button>,
  <button
    class="btn letter"
    data-testid="letter-button"
  >
    B
  </button>,
  <button
    class="btn letter"
    data-testid="letter-button"
    disabled=""
  >
    C
  </button>,
]
`;

exports[`<LetterBar /> matches snapshot with one 0 used letters 1`] = `
Array [
  <button
    class="btn letter"
    data-testid="letter-button"
  >
    A
  </button>,
  <button
    class="btn letter"
    data-testid="letter-button"
  >
    B
  </button>,
  <button
    class="btn letter"
    data-testid="letter-button"
  >
    C
  </button>,
]
`;
